workers/unity/Assets/Playground/Scripts/EntityInitialization/EntityGameObjectCreator.cs

```diff
@@ -1,46 +1,66 @@
-using System;
 using System.Collections.Generic;
-using Improbable.Worker;
-using Unity.Entities;
+using System.IO;
+using Generated.Improbable;
+using Generated.Improbable.Transform;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.GameObjectRepresentation;
 using UnityEngine;
+using Transform = Generated.Improbable.Transform.Transform;
 
 namespace Playground
 {
-    public class EntityGameObjectCreator
+    public class EntityGameObjectCreator : IEntityGameObjectCreator
     {
-        private readonly World world;
-        private readonly Dictionary<string, GameObject> cachedPrefabs;
+        private readonly Dictionary<string, Dictionary<string, GameObject>> cachedPrefabsPerWorkerType
+            = new Dictionary<string, Dictionary<string, GameObject>>();
 
-        public EntityGameObjectCreator(World world, Dictionary<string, GameObject> initialCachedPrefabs = null)
+        public GameObject CreateGameObjectForEntity(SpatialOSEntity entity, WorkerSystem worker)
         {
-            this.world = world;
-            cachedPrefabs = initialCachedPrefabs ?? new Dictionary<string, GameObject>();
-        }
+            if (!cachedPrefabsPerWorkerType.TryGetValue(worker.WorkerType, out var cachedPrefabs))
+            {
+                cachedPrefabsPerWorkerType[worker.WorkerType] = cachedPrefabs = new Dictionary<string, GameObject>();
+            }
 
-        public GameObject CreateEntityGameObject(Entity entity, string prefabPath, Vector3 position,
-            Quaternion rotation, EntityId spatialEntityId)
-        {
-            if (!cachedPrefabs.TryGetValue(prefabPath, out var prefab))
+            if (!entity.HasComponent<Metadata.Component>())
+            {
+                return null;
+            }
+
+            var prefabName = entity.GetComponent<Metadata.Component>().EntityType;
+            if (!entity.HasComponent<Transform.Component>())
             {
-                prefab = Resources.Load<GameObject>(prefabPath);
+                return cachedPrefabs[prefabName] = null;
+            }
+
+            var transform = entity.GetComponent<Transform.Component>();
+            var position = new Vector3(transform.Location.X, transform.Location.Y, transform.Location.Z) +
```

Comment:
Do we still not have conversion helpers to/from the Unity types?

Reply:
As far as I can tell, we don't, the existing code (in the transform sync module, and here in the game object initialization) all does it itself.

Reply:
I could add something as a driveby but I'm not really sure where it would belong either

Reply:
It doesn't need to happen in this PR, but let's at least make a ticket to make this nicer, since this is prone to typos, and it's gonna happen at some point.

Reply:
Done, UTY-1116.